docs/misp2_installation_manual_18.04.md

```diff
@@ -340,6 +342,77 @@ described in Section
 [5.3](#53-configuring-https-connection-between-misp2-application-and-x-road-security-server)
 of this guide.
 
+##### 4.3.2.1 Configuring support for the Estonian ID-card
+
+In order to configure the Estonian ID-card based authentication to work
+properly, please complete the following steps after the package has been
+installed (if you have upgraded from version 2.5.0 and had the ID-card
+previously set up, only step 6 is required):
+
+These steps require root privileges. These can be gained using the following
+command:
+
+```bash
+sudo -i
+```
+
+1. Download the required certificates from the Certificate Authority:
+
+    ```bash
+    cd /etc/apache2/ssl
+    wget -O sk_root_2011_crt.pem https://www.sk.ee/upload/files/EE_Certification_Centre_Root_CA.pem.crt
+    wget -O sk_root_2018_crt.pem https://c.sk.ee/EE-GovCA2018.pem.crt
+    wget -O sk_esteid_2011_crt.pem https://www.sk.ee/upload/files/ESTEID-SK_2011.pem.crt
+    wget -O sk_esteid_2015_crt.pem https://www.sk.ee/upload/files/ESTEID-SK_2015.pem.crt
+    wget -O sk_esteid_2018_crt.pem https://c.sk.ee/esteid2018.pem.crt
+    ```
+
+2. Create the `client_ca` folder under `/etc/apache2/ssl` install the
+   certificates there with the following commands:
+
+    ```bash
+    mkdir client_ca
+    cp -v sk_esteid_2011_crt.pem sk_esteid_2015_crt.pem sk_esteid_2018_crt.pem client_ca/
+    openssl x509 -addtrust clientAuth -trustout -in sk_esteid_2011_crt.pem -out sk_esteid_2011_client_auth_trusted_crt.pem
+    openssl x509 -addtrust clientAuth -trustout -in sk_esteid_2015_crt.pem -out sk_esteid_2015_client_auth_trusted_crt.pem
+    openssl x509 -addtrust clientAuth -trustout -in sk_esteid_2018_crt.pem -out sk_esteid_2018_client_auth_trusted_crt.pem
+    rm sk_esteid_2011_crt.pem sk_esteid_2015_crt.pem sk_esteid_2018_crt.pem
+    c_rehash client_ca/
+    ```
+
+3. Install the root certificates by running the following commands under
+   `/etc/apache2/ssl`:
+
+    ```bash
+    openssl x509 -addreject clientAuth -trustout -in sk_root_2011_crt.pem -out sk_root_2011_CA_trusted_crt.pem
+    openssl x509 -addreject clientAuth -trustout -in sk_root_2018_crt.pem -out sk_root_2018_CA_trusted_crt.pem
+    rm sk_root_2011_crt.pem sk_root_2018_crt.pem
+    ```
+
+4. Install the OCSP certificate by running the following command under
+   `/etc/apache2/ssl`:
+
+    ```bash
+    wget -O sk_esteid_ocsp.pem https://www.sk.ee/upload/files/SK_OCSP_RESPONDER_2011.pem.cer
+    ```
+
+5. Update the CRL and rehash Apache symbolic links under `ssl` by running the
+   following commands under `/etc/apache2/ssl`:
+
+    ```bash
+    ./updatecrl.sh "norestart"
+    c_rehash ./
+    ```
+
+6. Open the file `/etc/apache2/sites-enabled/ssl.conf` and change the parameter
+   `SSLCADNRequestPath` on line 164 to be the following (be sure to also remove
+   the `#` from the beginning so that it isn't commented out):
+
+    ```properties
+    SSLCADNRequestPath /etc/apache2/ssl/client_ca/
+    service apache2 restart
```

Comment:
For clarity, I'd add restarting Apache as a separate step:

7. After the configuration file is changed, Apache must be restarted using the command:

```bash
service apache2 restart
```

Reply:
Yep, separated it out to its own step.